doc/develop/test/twister.rst

```diff
@@ -375,6 +375,25 @@ harness: <string>
     Usually pertains to external dependency domains but can be anything such as
     console, sensor, net, keyboard, Bluetooth or pytest.
 
+platform_key: <list of platform attributes>
+    Often a test needs to only be built and run once to qualify as passing.
+    Imagine a library of code that depends on the platform architecture where
+    passing the test on a single platform for each arch is enough to qualify the
+    tests and code as passing. The platform_key attribute enables doing just
+    that.
+
+    For example to key on (arch, simulation) to ensure a test is run once
+    per arch and simulation (as would be most common)::
+
+      platform_key:
+        - arch
+        - simulation
+
```

Comment:
Can you provide a concrete example ?
When providing arch and simulation keys, will it be run once per arch and simulation or once per arch and once per simulation ? (I'm sure one of these 2 doesn't make sense, but let's make it clear in in reader's mind).
Also: `arch` is already used in documentation, so it's clear what it refers to.
But this is the first reference to `simulation` in twister doc. So it may be better to define it first.

Reply:
a platform_key forms a unique key into the hash table, so each attribute here is part of the key
e.g. test_key = sort([plat.arch, plat.simulation])

simulation exists already as a platform attribute, surprised its not documented. I can add that in another commit.

Reply:
> simulation exists already as a platform attribute, surprised its not documented. I can add that in another commit.

 Could not find it.

Reply:
See boards/arm/qemu_cortex_m3.yaml for example of where its used (simulation: qemu)

Reply:
Added a note in the docs about it